core/math/a_star.cpp

```diff
@@ -274,6 +275,20 @@ bool AStar::_solve(Point *begin_point, Point *end_point) {
 	return found_route;
 }
 
+float AStar::_estimate_cost(int p_from_id, int p_to_id) {
+    if (get_script_instance())
+        return get_script_instance()->call("_estimate_cost", p_from_id, p_to_id);
+        
+    return points[p_from_id]->pos.distance_to(points[p_to_id]->pos);
+}
+
+float AStar::_compute_cost(int p_from_id, int p_to_id) {
+    if (get_script_instance())
+        return get_script_instance()->call("_compute_cost", p_from_id, p_to_id);
```

Comment:
It should be done somewhat as here, I guess: https://github.com/godotengine/godot/blob/73eaf81c165e9379cee826a092dd612c4512c128/scene/gui/control.cpp#L574

(Note the use of SceneStringName too, it would make it an idea faster..)